apps/opik-backend/src/main/java/com/comet/opik/infrastructure/bi/ApplicationStartupListener.java

```diff
@@ -0,0 +1,147 @@
+package com.comet.opik.infrastructure.bi;
+
+import com.comet.opik.domain.IdGenerator;
+import com.comet.opik.infrastructure.OpikConfiguration;
+import com.comet.opik.infrastructure.lock.LockService;
+import com.google.inject.Injector;
+import jakarta.ws.rs.client.Client;
+import jakarta.ws.rs.client.ClientBuilder;
+import jakarta.ws.rs.client.Entity;
+import jakarta.ws.rs.core.MediaType;
+import jakarta.ws.rs.core.Response;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import reactor.core.publisher.Mono;
+import reactor.core.scheduler.Schedulers;
+import ru.vyarus.dropwizard.guice.module.lifecycle.GuiceyLifecycle;
+import ru.vyarus.dropwizard.guice.module.lifecycle.GuiceyLifecycleListener;
+import ru.vyarus.dropwizard.guice.module.lifecycle.event.GuiceyLifecycleEvent;
+import ru.vyarus.dropwizard.guice.module.lifecycle.event.InjectorPhaseEvent;
+
+import java.net.URI;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static com.comet.opik.infrastructure.lock.LockService.Lock;
+
+@Slf4j
+@RequiredArgsConstructor
+public class ApplicationStartupListener implements GuiceyLifecycleListener {
+
+    public static final String NOTIFICATION_EVENT_TYPE = "opik_os_startup_be";
+
+    // This event cannot depend on authentication
+    private final Client client = ClientBuilder.newClient();
+    private final AtomicReference<Injector> injector = new AtomicReference<>();
+
+    @Override
+    public void onEvent(GuiceyLifecycleEvent event) {
+
+        if (event instanceof InjectorPhaseEvent injectorEvent) {
+            injector.set(injectorEvent.getInjector());
+        }
```

Comment:
Probably a listener is too overkill for the purpose of what we want to do. It'd be good enough to just trigger the event from the application `run` method.

I'm afraid of altering the lifecycle and having to update this listener in the future for unrelated changes.

Specially, statements like setting this injection shouldn't be necessary.

Reply:
I'm not sure I understood this correctly. However, it's possible to have a listener to a particular event. This one is more generic because I could find a way to make a regular component in the application context. I believe adding it to the run method would make it more complex than it should

Reply:
I think the listener can be a very clean solution but:
1. Better extract the business logic to a service. The injections aren't great and it's hard to track dependencies such as DAOs etc. Then, the listener will just call the service to trigger the event. The service will handle all the inner business logic.
2. Great if it a listener for the particular event can be used. If not possible, after tacking 1 things should be much cleaner.